### PR TITLE
Add Tether permission to the Standard Permission

### PR DESCRIPTION
**Pull request by rahul1995**

### CODE REVIEW

File: cdap-proto/src/main/java/io/cdap/cdap/proto/security/StandardPermission.java [hunk 1]
```diff
@@ -52,6 +52,10 @@ public boolean isCheckedOnParent() {
    * Permission to delete specific entity.
    */
   DELETE,
+  /**
+   * Permission to tether other clusters with this CDAP instance
+   */
+  TETHER,
```
Comment (tivv) [inline cdap-proto/src/main/java/io/cdap/cdap/proto/security/StandardPermission.java]: Why is it StandardPermission? StandardPermission is for permission that is available for any resource. This looks like a special permission for some specific resource. Which resource is it going to be checked against?

Comment (rahul1995) [reply cdap-proto/src/main/java/io/cdap/cdap/proto/security/StandardPermission.java]: It is specifically for CDAP instance. Created a new class `InstancePermission`.

**Review by tivv — COMMENTED** [review]
Do we have an OK from @dli357  for this? Adding a new Permission should not be taken lightly as this require changes to all security extensions.
Overall, there are 3 ways of handling this:
 * Use Instance.Update - the most light way of doing this.
 * Add new EntityId - the most comprehensive way as we can fully control lifecycle
 * Add new permission - an intermediate one with significant cons.
Do we have pros and cons reviewed? Why Instance.Update is not enough?

**Review by dli357 — APPROVED** [review]
LGTM, thanks!

**Comment by rahul1995** [comment]
@tivv I believe the workflow will be that ODF will request for tethering. CDF will accept the request, and use ODF cluster to run pipelines. Hence ODF doesn't need to have update permission on CDF instance. That's why it makes sense to have a new tethering-related permission.
Also, didn't understand "new EntityId" approach that you mentioned. Could you please explain that?
Thanks.

**Review by tivv — APPROVED** [review]
Chatted sepearately about use-case will confirm with Dennis later, good to go now

**Pull request merged**